src/api/contract/clusterprovider/v1.ts

```diff
@@ -0,0 +1,34 @@
+// This module is contractual and should not be changed after release.
```

Comment:
any reason why these contract files are baked into the extension rather than being distributed by the api node module?

Reply:
The extension needs copies of the contracts so it can check its implementation conforms to the contracts.  It could do this by importing the NPM module, but then we'd have a weird sort-of-circular-dependency situation which, although it _could_ be made to work, would be wearisome.  (Think of the workflow where you'd have to write and pack the NPM module _but not publish it_ then amend the reference in the extension to the module TGZ, then implement the contracts, then amend the reference back to an _unpublished_ version of the NPM module, push and merge the extension in a nonbuilding state (because it couldn't find the NPM module with the updated contracts), then finally publish the NPM module.)